packages/host/app/components/operator-mode.gts

```diff
@@ -81,21 +92,126 @@ export default class OperatorMode extends Component<Signature> {
+  }
+
+  @action async save(card: Card) {
+    await this.saveCardFieldValues(card);
+    this.write.perform(card);
+  }
+
+  private write = restartableTask(async (card: Card) => {
+    let updatedCard = await this.cardService.saveModel(card);
+    this.formats.set(updatedCard, 'isolated');
   });
 
+  private async saveCardFieldValues(card: Card) {
+    let fields = await this.cardService.getFields(card);
+    for(let fieldName of Object.keys(fields)) {
+      if (fieldName === 'id') continue;
+
+      let field = fields[fieldName];
+      if ((field.fieldType === 'contains' ||  field.fieldType === 'containsMany') && !(await this.cardService.isPrimitive(field.card))) {
+        await this.saveCardFieldValues((card as any)[fieldName]);
+      }
+
+      let cardFieldValue = this.cardFieldValues.get(card);
+      if (!cardFieldValue) {
+        cardFieldValue = new Map<string, any>;
+      }
+      cardFieldValue.set(fieldName, (card as any)[fieldName]);
+      this.cardFieldValues.set(card, cardFieldValue);
+    }
+  }
+
+  private async rollbackCardFieldValue(card: Card) {
+    let fields = await this.cardService.getFields(card);
+    for(let fieldName of Object.keys(fields)) {
+      if (fieldName === 'id') continue;
+
+      let field = fields[fieldName];
+      if ((field.fieldType === 'contains' ||  field.fieldType === 'containsMany') && !(await this.cardService.isPrimitive(field.card))) {
+        await this.rollbackCardFieldValue((card as any)[fieldName]);
+      }
+
+      let cardFieldValue = this.cardFieldValues.get(card);
+      if (cardFieldValue) {
+        (card as any)[fieldName] = cardFieldValue.get(fieldName);
+      }
+    }
+  }
+
   <template>
     <div class='operator-mode-desktop-overlay'>
       <CardCatalogModal />
       <CreateCardModal />
-
       <div class='operator-mode-card-stack'>
         {{#each this.stack as |card|}}
-          <div class='operator-mode-stack-card'>
-            <card />
+          <div class='operator-mode-card-stack__card'>
+            <div class={{cn 'operator-mode-card-stack__card__item' operator-mode-card-stack__card__item_edit=(eq (getValueFromWeakMap this.formats card) 'edit')}}>
+              <Preview @card={{card}} @format={{this.getFormat card}} />
+            </div>
+            <div class='operator-mode-card-stack__card__header'>
+              {{#if (not (eq (getValueFromWeakMap this.formats card) 'edit'))}}
+                <IconButton
+                  @icon='icon-horizontal-three-dots'
+                  @width='20px'
+                  @height='20px'
+                  class='icon-button'
+                  aria-label='Edit'
+                  {{on 'click' (fn this.edit card)}}
+                />
+              {{/if}}
+              <IconButton
+                @icon='icon-x'
+                @width='20px'
+                @height='20px'
+                class='icon-button'
+                aria-label='Edit'
+                {{on 'click' (fn this.edit card)}}
```

Comment:
I'm confused. If this is the X icon, why would the aria-label and click handler be "edit"?

Reply:
nice catch! Thanks